flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/table/ParameterizedPredicatePlanTest.java

```diff
@@ -0,0 +1,169 @@
+                                .getPredicate()
+                                .charAt(
+                                        thisParameterizedPredicateEquals
+                                                .getIndexesOfPredicatePlaceHolders()
+                                                .get(0)))
+                .isEqualTo(
+                        JdbcFilterPushdownPreparedStatementVisitor.PUSHDOWN_PREDICATE_PLACEHOLDER
+                                .charAt(0));
+
+        ParameterizedPredicate thatParameterizedPredicateGTE =
+                new ParameterizedPredicate("`type2`");
+        thatParameterizedPredicateGTE.combine(
+                JdbcFilterPushdownPreparedStatementVisitor.OPERATOR_GREATER_THAN_OR_EQUAL,
+                parameterizedPredicatePlaceHolder);
+        assertThat(
+                        thatParameterizedPredicateGTE
+                                .getPredicate()
+                                .charAt(
+                                        thatParameterizedPredicateGTE
+                                                .getIndexesOfPredicatePlaceHolders()
+                                                .get(0)))
+                .isEqualTo(
+                        JdbcFilterPushdownPreparedStatementVisitor.PUSHDOWN_PREDICATE_PLACEHOLDER
+                                .charAt(0));
+
+        thisParameterizedPredicateEquals.combine(
+                JdbcFilterPushdownPreparedStatementVisitor.OPERATOR_OR,
+                thatParameterizedPredicateGTE);
+        assertThat(thisParameterizedPredicateEquals.getPredicate())
+                .isEqualTo("((`type1` = ?) OR (`type2` >= ?))");
+        assertThat(thisParameterizedPredicateEquals.getIndexesOfPredicatePlaceHolders().size())
+                .isEqualTo(2);
+        assertThat(
+                        thisParameterizedPredicateEquals
+                                .getPredicate()
+                                .charAt(
+                                        thisParameterizedPredicateEquals
+                                                .getIndexesOfPredicatePlaceHolders()
+                                                .get(0)))
+                .isEqualTo(
+                        JdbcFilterPushdownPreparedStatementVisitor.PUSHDOWN_PREDICATE_PLACEHOLDER
+                                .charAt(0));
+        assertThat(
+                        thisParameterizedPredicateEquals
+                                .getPredicate()
+                                .charAt(
+                                        thisParameterizedPredicateEquals
+                                                .getIndexesOfPredicatePlaceHolders()
+                                                .get(1)))
+                .isEqualTo(
+                        JdbcFilterPushdownPreparedStatementVisitor.PUSHDOWN_PREDICATE_PLACEHOLDER
+                                .charAt(0));
+    }
+
+    @Test
+    public void testCombineAll() {
```

Comment:
Looks like a very complicated test
The main drawback is that if there is an issue with some predicate in the middle there is no way to debug it without replaying all previous ... imho parameterized tests would be better

Reply:
I was hoping to test the order of predicates, a long ish query and ensure we hit every predicate.

 I can look at a parameterized alternative; are you thinking of ORs testing permutations of binary op then unitary op, unitary op then binary op, binary op then binary op. Then repeat with binary operations having the literal on the left. Then more complex tests with nested ANDs, ORs and brackets, each of which would need a separate test. I wonder if I could do the parameterized tests in a subsequent pr. For this pr I could add some cases for the complex tests. fyi I am not able to work too much on this until Thursday. I hope to put something up before then, but will leave this conversation open, pending your response.

Reply:
Please have a look at https://github.com/davidradl/flink-connector-jdbc/pull/1 where there is a proposal with parameterized tests